### PR TITLE
Creates preliminary landing pages for new Beam website layout.

### PR DESCRIPTION
**Pull request by devin-donnelly**
Creates the following directory structure:
docs/
docs/use
docs/learn
docs/contribute

Creates index.md landing pages in each directory to make the docs and sections more findable and provide guidance and explanation about what to read.

### CODE REVIEW

**Comment by francesperry** [comment]
@jbonofre Just FYI JB -- should presumably merge in just fine with your design work on the main page.

**Comment by francesperry** [comment]
Can we update the header in this PR too?

File: docs/learn/programming-guide.md [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+  * [Side Inputs and Side Outputs](#transforms-sideio)
+* [I/O](#io)
+* [Running the Pipeline](#running)
+* [Data Encoding and Type Safety](#coders)
+* [Working with Windowing](#windowing)
+* [Working with Triggers](#triggers)
+
+## <a name="#overview"></a>Overview
+
+To use Beam, you need to first create a driver program using the classes in one of the Beam SDKs. Your driver program *defines* your pipeline, including all of the inputs, transforms, and outputs; it also sets execution options for your pipeline, such as the Pipeline Runner (which, in turn, determines what back-end your pipeline will run on).
```
Comment (francesperry) [inline docs/learn/programming-guide.md]: Might rephrase -- right now this implies the driver program has to set the runner, whereas we often recommend you do it via a commandline flag.

File: docs/learn/programming-guide.md [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+  <div class="alert alert-info alert-dismissible" role="alert">
+  <span class="glyphicon glyphicon-flag" aria-hidden="true"></span>
+  <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+  The Apache Beam project is in the process of bootstrapping. This includes the creation of project resources, the refactoring of the initial code submission, and the formulation of project documentation, planning, and design documents. For more information about Beam see the <a href="/getting_started/">getting started page</a>.
+  </div>
+</p>
+
+# Beam Programming Guide
+
+The **Beam Programming Guide** is intended for Beam users who want to use the Beam SDKs to create data processing pipelines. It contains guidance for using the Beam SDK classes to build and test your pipeline. It is not intended as an exhaustive reference, but as a higher-level guide to programmatically building your Beam pipeline.
```
Comment (francesperry) [inline docs/learn/programming-guide.md]: I'd vote for a more explicit disclaimer that this is a language-independent guide. (And at some point a mention of toggling code snippets?) May also want to point to the SDKs landing page for things like JavaDoc (once we create it ;)

File: docs/learn/programming-guide.md (hunk 1 above)
Comment (devin-donnelly) [inline docs/learn/programming-guide.md]: Done.

File: docs/learn/programming-guide.md (hunk 2 above)
Comment (devin-donnelly) [inline docs/learn/programming-guide.md]: Done.

File: _includes/header.html [hunk 3]
```diff
@@ -8,33 +8,52 @@
...
-          </ul>
-        </li>
+		  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Use <span class="caret"></span></a>
+		  <ul class="dropdown-menu">
+			  <li><a href="{{ site.baseurl }}/docs/use/">User Hub</a></li>
+			  <li role="separator" class="divider"></li>
+			  <li><a href="{{ site.baseurl }}/docs/use">Beam Overview</a></li>
+			  <li><a href="{{ site.baseurl }}/getting_started/">Quickstart</a></li>
+			  <li class="dropdown-header">Example Walkthroughs</li>
+			  <li><a href="{{ site.baseurl }}/docs/walkthroughs/wordcount">WordCount</a></li>
```
Comment (francesperry) [inline _includes/header.html]: link 404s

File: _includes/header.html [hunk 4]
```diff
@@ -8,33 +8,52 @@
...
+			  <li><a href="{{ site.baseurl }}/docs/walkthroughs/mobile-gaming">Mobile Gaming</a></li>
```
Comment (francesperry) [inline _includes/header.html]: link 404s

File: _includes/header.html [hunk 5]
```diff
@@ -8,33 +8,52 @@
...
+			  <li role="separator" class="divider"></li>
+			  <li><a href="{{ site.baseurl }}">Release Information</a></li>
+			  <li class="dropdown-header">Support Resources</li>
+			  <li><a href="{{ site.baseurl }}/mailing_lists/">Mailing Lists</a></li>
+              <li><a href="{{ site.baseurl }}/issue_tracking/">Issue Tracking</a></li>
+			  <li><a href="http://stackoverflow.com">Beam on StackOverflow</a></li>
```
Comment (francesperry) [inline _includes/header.html]: link to stackoverflow.com/questions/tagged/apache-beam

**Pull request closed without merging**